From 57d8e450eb5f9500d9136374667e62dd1ee99ff2 Mon Sep 17 00:00:00 2001
From: Varun Date <date.varun123@gmail.com>
Date: Fri, 13 May 2016 13:36:27 +0530
Subject: [PATCH] Settings: Move Advanced Reboot to Button Settings

project packages/apps/Settings/
diff --git a/packages/apps/Settings/res/xml/button_settings.xml b/packages/apps/Settings/res/xml/button_settings.xml
index ffb8c780c..1ebb14f40 100644
--- a/packages/apps/Settings/res/xml/button_settings.xml
+++ b/packages/apps/Settings/res/xml/button_settings.xml
@@ -89,6 +89,11 @@
             android:summary="@string/camera_double_tap_power_gesture_desc"
             android:persistent="false" />
 
+        <SwitchPreference
+            android:key="advanced_reboot"
+            android:title="@string/advanced_reboot_title"
+            android:summary="@string/advanced_reboot_summary" />
+
     </PreferenceCategory>
 
     <PreferenceCategory
diff --git a/packages/apps/Settings/res/xml/development_prefs.xml b/packages/apps/Settings/res/xml/development_prefs.xml
index 0e939e002..5c4dcc90f 100644
--- a/packages/apps/Settings/res/xml/development_prefs.xml
+++ b/packages/apps/Settings/res/xml/development_prefs.xml
@@ -32,11 +32,6 @@
             android:title="@*android:string/bugreport_title"
             android:dialogTitle="@*android:string/bugreport_title" />
 
-    <SwitchPreference
-        android:key="advanced_reboot"
-        android:title="@string/advanced_reboot_title"
-        android:summary="@string/advanced_reboot_summary" />
-
     <PreferenceScreen
             android:key="local_backup_password"
             android:title="@string/local_backup_password_title"
diff --git a/packages/apps/Settings/src/com/android/settings/ButtonSettings.java b/packages/apps/Settings/src/com/android/settings/ButtonSettings.java
index d4d44cbab..b1da504c5 100644
--- a/packages/apps/Settings/src/com/android/settings/ButtonSettings.java
+++ b/packages/apps/Settings/src/com/android/settings/ButtonSettings.java
@@ -29,6 +29,7 @@ import android.content.SharedPreferences.Editor;
 import android.os.Bundle;
 import android.os.RemoteException;
 import android.os.Handler;
+import android.os.UserHandle;
 import android.preference.ListPreference;
 import android.preference.Preference;
 import android.preference.PreferenceCategory;
@@ -84,6 +85,7 @@ public class ButtonSettings extends SettingsPreferenceFragment implements
     private static final String KEY_VOLUME_CONTROL_RING_STREAM = "volume_keys_control_ring_stream";
     private static final String KEY_CAMERA_DOUBLE_TAP_POWER_GESTURE
             = "camera_double_tap_power_gesture";
+    private static final String ADVANCED_REBOOT_KEY = "advanced_reboot";
 
     private static final String KEY_NAVIGATION_BAR_HEIGHT = "navigation_bar_height";
 
@@ -142,6 +144,7 @@ public class ButtonSettings extends SettingsPreferenceFragment implements
     private SwitchPreference mPowerEndCall;
     private SwitchPreference mHomeAnswerCall;
     private SwitchPreference mCameraDoubleTapPowerGesture;
+    private SwitchPreference mAdvancedReboot;
 
     private PreferenceCategory mNavigationPreferencesCat;
 
@@ -211,6 +214,11 @@ public class ButtonSettings extends SettingsPreferenceFragment implements
 
         mHandler = new Handler();
 
+        mAdvancedReboot = (SwitchPreference) findPreference(ADVANCED_REBOOT_KEY);
+        mAdvancedReboot.setChecked(MKSettings.Secure.getIntForUser(resolver,
+            MKSettings.Secure.ADVANCED_REBOOT, 1, UserHandle.USER_CURRENT) == 1);
+        mAdvancedReboot.setOnPreferenceChangeListener(this);
+
         // Force Navigation bar related options
         mDisableNavigationKeys = (SwitchPreference) findPreference(DISABLE_NAV_KEYS);
 
@@ -637,6 +645,11 @@ public class ButtonSettings extends SettingsPreferenceFragment implements
             Settings.Secure.putInt(getContentResolver(), CAMERA_DOUBLE_TAP_POWER_GESTURE_DISABLED,
                     value ? 0 : 1 /* Backwards because setting is for disabling */);
             return true;
+        } else if (preference == mAdvancedReboot) {
+            boolean show = (Boolean) newValue;
+            MKSettings.Secure.putIntForUser(getActivity().getContentResolver(),
+                    MKSettings.Secure.ADVANCED_REBOOT, show ? 1 : 0, UserHandle.USER_CURRENT);
+            return true;
         }
         return false;
     }
diff --git a/packages/apps/Settings/src/com/android/settings/DevelopmentSettings.java b/packages/apps/Settings/src/com/android/settings/DevelopmentSettings.java
index 79f9a386e..1a31c8ea7 100644
--- a/packages/apps/Settings/src/com/android/settings/DevelopmentSettings.java
+++ b/packages/apps/Settings/src/com/android/settings/DevelopmentSettings.java
@@ -196,8 +196,6 @@ public class DevelopmentSettings extends SettingsPreferenceFragment
 
     private static final String DEVELOPMENT_TOOLS = "development_tools";
 
-    private static final String ADVANCED_REBOOT_KEY = "advanced_reboot";
-
     private static final String DEVELOPMENT_SHORTCUT_KEY = "development_shortcut";
 
     private static final int RESULT_DEBUG_APP = 1000;
@@ -295,8 +293,6 @@ public class DevelopmentSettings extends SettingsPreferenceFragment
 
     private Preference mRootAppops;
 
-    private SwitchPreference mAdvancedReboot;
-
     private SwitchPreference mDevelopmentShortcut;
 
     private SwitchPreference mColorTemperaturePreference;
@@ -380,7 +376,6 @@ public class DevelopmentSettings extends SettingsPreferenceFragment
         mDebugViewAttributes = findAndInitSwitchPref(DEBUG_VIEW_ATTRIBUTES);
         mPassword = (PreferenceScreen) findPreference(LOCAL_BACKUP_PASSWORD);
         mAllPrefs.add(mPassword);
-        mAdvancedReboot = findAndInitSwitchPref(ADVANCED_REBOOT_KEY);
 
         mDevelopmentShortcut = findAndInitSwitchPref(DEVELOPMENT_SHORTCUT_KEY);
 
@@ -390,7 +385,6 @@ public class DevelopmentSettings extends SettingsPreferenceFragment
             disableForUser(mClearAdbKeys);
             disableForUser(mEnableTerminal);
             disableForUser(mPassword);
-            disableForUser(mAdvancedReboot);
             disableForUser(mDevelopmentShortcut);
         }
 
@@ -746,24 +740,12 @@ public class DevelopmentSettings extends SettingsPreferenceFragment
         updateSimulateColorSpace();
         updateUSBAudioOptions();
         updateRootAccessOptions();
-        updateAdvancedRebootOptions();
         updateDevelopmentShortcutOptions();
         if (mColorTemperaturePreference != null) {
             updateColorTemperature();
         }
     }
 
-    private void writeAdvancedRebootOptions() {
-        MKSettings.Secure.putInt(getActivity().getContentResolver(),
-                MKSettings.Secure.ADVANCED_REBOOT,
-                mAdvancedReboot.isChecked() ? 1 : 0);
-    }
-
-    private void updateAdvancedRebootOptions() {
-        mAdvancedReboot.setChecked(MKSettings.Secure.getInt(getActivity().getContentResolver(),
-                MKSettings.Secure.ADVANCED_REBOOT, 0) != 0);
-    }
-
     private void resetDevelopmentShortcutOptions() {
         MKSettings.Secure.putInt(getActivity().getContentResolver(),
                 MKSettings.Secure.DEVELOPMENT_SHORTCUT, 0);
@@ -2047,8 +2029,6 @@ public class DevelopmentSettings extends SettingsPreferenceFragment
             writeColorTemperature();
         } else if (preference == mUSBAudio) {
             writeUSBAudioOptions();
-        } else if (preference == mAdvancedReboot) {
-            writeAdvancedRebootOptions();
         } else if (INACTIVE_APPS_KEY.equals(preference.getKey())) {
             startInactiveAppsFragment();
         } else if (preference == mDevelopmentShortcut) {
