
project vendor/cm/
diff --git a/vendor/cm/config/common.mk b/vendor/cm/config/common.mk
index 5ad739a9..5291497f 100644
--- a/vendor/cm/config/common.mk
+++ b/vendor/cm/config/common.mk
@@ -75,10 +75,6 @@ PRODUCT_COPY_FILES += \
 PRODUCT_COPY_FILES += \
     vendor/cm/Features.mkdn:system/etc/RR/Features.txt
 
-# NexusLauncher
-PRODUCT_COPY_FILES += \
-    vendor/cm/prebuilt/NexusLauncher/NexusLauncher.apk:system/app/NexusLauncher/NexusLauncher.apk
-
 # Wallpaper
 PRODUCT_COPY_FILES += \
     vendor/cm/prebuilt/Wallpaper/Wallpaper.apk:system/app/Wallpaper/Wallpaper.apk
